### DISCUSSION TITLE
about custom DQL function get error

### DISCUSSION THREAD

**Discussion by langziyang** [doctrine/doctrine2#11120] [Support Questions]
hi,i am use symfony, i want learn custom DQL function,first i use querybuilder ->andWhere('YEAR(a.createdAt)=2023').
then i register myfunction to datetime_functions and it`s working fine.
so i guess i am genius....until i want custom JSON_CONTAINS function

```
$qb = $this->createQueryBuilder('a');
        $qb->where('JSON_CONTAINS(a.roles,:role)')
            ->setParameter('role','"ROLE_ADMIN"');

        return $qb->getQuery()->getResult();
```
so i get 
```
[AssertionError](http://192.168.100.13/#trace-box-1)
HTTP 500 Internal Server Error
assert($token !== null)
```
even i delete all custom functin file and all register code.

why and how to fix it?thanks

**Comment by mtarld** [comment]
You need to specify a value to compare `JSON_CONTAINS` with.
Therefore, `JSON_CONTAINS(a.roles,:role) = true` might work here
